PythonCodeConventions.md

```diff
@@ -1,24 +1,33 @@
 Python Code Conventions
 =======
 
-PEP8 is the code style guide for Python code used by the main python distribution.
-This style guide can be found at:
-https://www.python.org/dev/peps/pep-0008/
+General
+---
+Ultimaker adheres to the PEP8 coding guidelines as described here: https://www.python.org/dev/peps/pep-0008/,
+with the following adjustments, exceptions and specific choices:
 
-Our coding conventions follow the PEP style guide, with the exceptions stated in the generic code_conventions.md provided in this repository. This also means that for rules regarding class naming, documentation, etc., the rules described in the general code convention document MUST be used.
-
-For convenience, these are re-listed here. In case of conflict, the generic code_conventions are considered to be leading.
-* Always use spaces. While PEP8 allows tabs for backwards compatibility, we do not.
-* Strings are double quotes. While Python allows single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
-* "Maximum line length" - PEP8 specifies a maximum line length of 79 characters. At Ultimaker we have no official line length limit, although we strive to keep lines below 120 characters.
-* Naming convention:
-  * Function names in PEP8 are in lower_case_with_underscores. At Ultimaker we decided to match our C++ coding convention and use mixedCase (CapitalizedWords by initial lowercase character).
-  * Parameters, variables and members follow the lower_case_with_underscores convention. PEP8 has no recommendation on this.
+* Always use (4) spaces instead of tabs.
+* Strings use double quotes to match C++ style.
+* Maximum line length is set to 120 characters. See the 'line breaking' sectoin below for specific suggested PEP8 style choices.
+* String formatting, multiple elements:
+``` python
+"{foo}_{bar}".format(foo="foo", bar="bar")
```

Comment:
Hm, I'll discuss this with (some members of) the team since this differs from PEP8 but is not explicitly mentioned in the list of exceptions. Taken from PEP8:

`
Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.

Yes:

def complex(real, imag=0.0):
    return magic(r=real, i=imag)
No:

def complex(real, imag = 0.0):
    return magic(r = real, i = imag)
`

Also the "code_conventions.md" file is IMHO not a very good guide:
* Tries to copy PEP8, but is far from complete. Where I think we should not copy PEP8 at all, just the exceptions and specific choices we make therein.
* It is filled with TODOs and gaps.

Reply:
Spaces were not discussed, team was under the impression we were using PEP8 on this. Will have to await next meeting.

Reply:
It has gasp todos that are also not handled in this document, so that shouldn't be the point. 

Going back onto discussion; My dyslexia really likes proper room around parts of sentences. This means that it chokes up on things like some_variable_that_we_need="someReallyLongButExplicitThing".

This is why I argued back in the day that we should break with PEP on that regard (It not being part of the python code conventions is intentional!)

Reply:
The intention of this document was not to fill all gaps or missing guides. It addresses a few specific cases as discussed in our team's architecture meeting.

Doesn't matter what we do concerning the spaces, I was just relating to how it got to be like it is in this PR. Doesn't matter what we use, as long as we stick to it :) I can see how not using spaces, make reading harder for someone with dyslexia. Point made, I will 'report' that back to the team and adjust examples accordingly.

Reply:
@nallath @Ghostkeeper added spaces, removed code_convention overlapping items. Will update/expand code_conventions.md where applicable.